src/bin/pgcopydb/cli_common.c

```diff
@@ -228,92 +228,33 @@ cli_copydb_getenv(CopyDBOptions *options)
 		}
 	}
 
-	if (env_exists(PGCOPYDB_TABLE_JOBS))
-	{
-		char jobs[BUFSIZE] = { 0 };
 
-		if (get_env_copy(PGCOPYDB_TABLE_JOBS, jobs, sizeof(jobs)))
-		{
-			if (!stringToInt(jobs, &options->tableJobs) ||
-				options->tableJobs < 1 ||
-				options->tableJobs > 128)
-			{
-				log_fatal("Failed to parse PGCOPYDB_TABLE_JOBS: \"%s\"",
-						  jobs);
-				++errors;
-			}
-		}
-		else
-		{
-			/* errors have already been logged */
-			++errors;
-		}
-	}
-
-	if (env_exists(PGCOPYDB_INDEX_JOBS))
-	{
-		char jobs[BUFSIZE] = { 0 };
-
-		if (get_env_copy(PGCOPYDB_INDEX_JOBS, jobs, sizeof(jobs)))
-		{
-			if (!stringToInt(jobs, &options->indexJobs) ||
-				options->indexJobs < 1 ||
-				options->indexJobs > 128)
-			{
-				log_fatal("Failed to parse PGCOPYDB_INDEX_JOBS: \"%s\"",
-						  jobs);
-				++errors;
-			}
-		}
-		else
-		{
-			/* errors have already been logged */
-			++errors;
-		}
-	}
+	EnvParser parsers[] = {
+		{ PGCOPYDB_TABLE_JOBS, ENV_TYPE_INT, &(options->tableJobs), 1, 128 },
+		{ PGCOPYDB_INDEX_JOBS, ENV_TYPE_INT, &(options->indexJobs), 1, 128 },
```

Comment:
How are you going to share that parts at two different places? I would host the target value within the structure, like we do in `parseSingleValueResult()` and `SingleValueResultContext`, and then have the caller code dispatch the parsed value at the right place. 

That way this array can be defined only once, and be the authoritative list of supported environment variables.

Reply:
I was planning to have a copy of this array with different target pointer values. Your suggestion is definitely an improvement on my current implementation.

Reply:
Now that I spent some more time on this task, I realize only 2 items in this list makes sense for list commands: `splitMaxParts` and `estimateTableSizes`. It made more sense for me to create a new array with 2 items in `cli_list.c` instead of improving the structure to be reusable in two places.

Reply:
Makes sense, I was under the impression that more was needed from this place.